tutorials/time-freq/ssvep.py

```diff
@@ -0,0 +1,382 @@
+# Calculate SNR
+# ^^^^^^^^^^^^^
+
+stim_freq = 12.
+noise_n_neighborfreqs = 3
+noise_skip_neighborfreqs = 1
+snrs = snr_spectrum(psds, noise_n_neighborfreqs=noise_n_neighborfreqs,
+                    noise_skip_neighborfreqs=noise_skip_neighborfreqs)
+
+###############################################################################
+# Find frequency bin containing stimulation frequency
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Ideally, this bin should should have the stimulation frequency exactly in the
+# center.
+
+tmp_distlist = abs(np.subtract(freqs, stim_freq))
+i_signal = np.where(tmp_distlist == min(tmp_distlist))[0][0]
+# could be updated to support multiple frequencies
+
+###############################################################################
+# Calculate SNR
+# ^^^^^^^^^^^^^
+# Extract and average SNRs at this frequency
+snrs_stim = snrs[:, :, i_signal]  # trial subselection can be done here
+print('average SNR at %iHz (all channels, all trials): %.3f '
+      % (stim_freq, snrs_stim.mean()))
+
+
+###############################################################################
+# Visualization
+# -------------
+
+
+##############################################################################
+# Plot power spectral density
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# code snippet from
+# https://martinos.org/mne/stable/auto_examples/time_frequency/plot_compute_raw_data_spectrum.html  # noqa E501
+
+fig, axes = plt.subplots(1, 1, sharex='all', sharey='all', dpi=300)
+rng = range(np.where(np.floor(freqs) == 1.)[0][0],
+            np.where(np.ceil(freqs) == fmax - 1)[0][0])
+
+psds_plot = 10 * np.log10(psds)
+psds_mean = psds_plot.mean((0, 1))[rng]
+psds_std = psds_plot.std((0, 1))[rng]
+axes.plot(freqs[rng], psds_mean, color='b')
+axes.fill_between(freqs[rng], psds_mean - psds_std, psds_mean + psds_std,
+                  color='b', alpha=.5)
+axes.set(title="PSD spectrum", xlabel='Frequency [Hz]',
+         ylabel='Power Spectral Density [dB]')
+plt.xlim([0, fmax])
+fig.show()
+
+##############################################################################
+# SNR spectrum - trial average
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+fig, axes = plt.subplots(1, 1, sharex='all', sharey='all', dpi=300)
+# Average over trials
+axes.plot(freqs, snrs.mean(axis=0).T, color='b')
+# Average over trials and channels
+axes.plot(freqs, snrs.mean(axis=0).mean(axis=0), color='r')
+axes.set(title="SNR spectrum - trial average", xlabel='Frequency [Hz]',
+         ylabel='SNR', ylim=[0, 20])
+fig.show()
+
+##############################################################################
+# SNR spectrum - channel average
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+fig, axes = plt.subplots(1, 1, sharex='all', sharey='all', dpi=300)
+# Average over channels
+axes.plot(freqs, snr_spectrum(psds[:, :, :], 3, 1).mean(axis=1).T, color='b')
+# Average over trials and channels
+axes.plot(freqs, snr_spectrum(psds[:, :, :], 3, 1).mean(axis=0).mean(axis=0),
```

Comment:
i think for the SNR plots, stem plots are actually more appropriate. 
the only thing to change would be `axes.plot()` -> `axes.stem()`

what do you think, would you try it out?

Reply:
Totally agree that line plots suck here.

Stem plots look good for one spectrum. When there are a bunch of them (like one for each channel/trial in our case) they don't work that well - matplotlib won't even allow you to plot multiple spectra at once. I can try to plot stem plots for each row in a loop using a light hue or adding transparency and then doing a stem plot of the average using the saturated/opaque coloring. Might be good to switch to scatter plots for individual spectra and use a stem plot for the average.

We can play with the plots later though. Let's have a fully working tutorial without errors first. Then we can refactor code, update plots, add more explanations and/or ways to reduce the dimensionality, etc. What do you think?